viewer/packages/data-providers/src/image-360-data-providers/Cdf360ImageEventProvider.ts

```diff
@@ -332,5 +334,27 @@ export class Cdf360ImageEventProvider implements Image360Provider<Metadata> {
     }
   }
 
-  public async get360ImageFaces(): Promise<void> {}
+  public async get360ImageAssets(image360FileDescriptors: Image360FileDescriptor[]): Promise<IdEither[]> {
+    const fileIds = image360FileDescriptors.map(desc => desc.fileId);
+    const assetListPromises = chunk(fileIds, 1000).map(async idList => {
+      const annotationArray = await this._client.annotations
+        .list({
+          filter: {
+            annotatedResourceIds: idList.map(id => ({ id })),
+            annotatedResourceType: 'file',
+            annotationType: 'images.AssetLink'
+          }
+        })
+        .autoPagingToArray();
+      const assetIds = annotationArray.map(
+        annotation => (annotation.data as AnnotationsCogniteAnnotationTypesImagesAssetLink).assetRef
+      );
+
+      return assetIds;
+    });
+
+    const assetIds = (await Promise.all(assetListPromises)).flat() as IdEither[];
```

Comment:
Should typeguard here and throw with proper error message if this fails.

Reply:
Instead used `filter()` to only admit elements satisfying `IdEither`.